### DISCUSSION TITLE
Use proper logger for backend API to make it configurable

### DISCUSSION THREAD

**Discussion by code-haven** [langfuse/langfuse#1073] [Ideas]
Hey there, we notice that langfuse is generating a lot of log volume by outputting raw events in stdout. Is there a way for us to turn off raw event logging at these lines? (at-least in our production environment)

1. `handling ingestion event` https://github.com/langfuse/langfuse/blob/d94e880520bda84d3514a33545a9c4aee8db1da9/src/pages/api/public/ingestion.ts#L146
2. `handling single event`: https://github.com/langfuse/langfuse/blob/d94e880520bda84d3514a33545a9c4aee8db1da9/src/pages/api/public/ingestion.ts#L210

Thanks!

**Comment by marcklingen** [comment]
Currently, it is not configurable because it uses `console.log`. I agree that it would be helpful to have the ability to configure this.

Do you have an implementation in mind and would you be interested in contributing to this?

**Comment by code-haven** [comment]
Not a nodejs expert but I imagine we'd need to use a proper logger like https://getpino.io/#/ and change all `console.log(...)` statements to something like `logger.debug(...)`. The log **"level"** should be configurable by the user. It can default to **INFO** if `NODE_ENV = production`.

**Comment by marcklingen** [comment]
Turning this into an idea post as this is a great feature request

**Comment by lbatteau** [comment]
Has this been implemented or is it planned? Very much needed, as Langfuse would be leaking everything into our logs.

**Reply by lbatteau** [reply]
It looks like it was added [here](https://github.com/langfuse/langfuse/pull/1478/files), but I don't see a way to set the level.

**Comment by marcklingen** [comment]
Update on this. We experimented with a number of loggers and approaches to get this right across nextjs (painful) and express. 

@Steffen911 added this today based on winston. You can configure logging via `LANGFUSE_LOG_FORMAT` and `LANGFUSE_LOG_LEVEL`, see docs for details: https://langfuse.com/docs/deployment/self-host#configuring-environment-variables
